src/main/java/org/openstreetmap/atlas/geography/atlas/builder/RelationBean.java

```diff
@@ -49,6 +52,21 @@ public RelationBeanItem(final RelationBeanItem item)
             this.type = item.type;
         }
 
+        @Override
+        public int compareTo(final RelationBeanItem other)
+        {
+            int result = this.getType().compareTo(other.getType());
+            if (result == 0)
+            {
+                result = Long.compare(this.getIdentifier(), other.getIdentifier());
+            }
+            if (result == 0)
+            {
+                result = this.getRole().compareTo(other.getRole());
```

Comment:
is this even possible?

Reply:
If you compare an item to itself, yes. But I think your question applies more to the same relation, and unfortunately yes too, since OSM does not enforce removing duplicates. This is why we keep lists along, instead of sets!

Reply:
I see, interesting!